lib/Conversion/TosaToSecretArith/TosaToSecretArith.cpp

```diff
@@ -0,0 +1,168 @@
+#include "llvm/include/llvm/Support/ErrorHandling.h"  // from @llvm-project
+#include "llvm/include/llvm/Support/LogicalResult.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlow/ConstantPropagationAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlow/DeadCodeAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlowFramework.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"    // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tosa/IR/TosaOps.h"   // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinAttributes.h"      // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinOps.h"             // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinTypes.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"   // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/OpDefinition.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/TypeUtilities.h"          // from @llvm-project
+#include "mlir/include/mlir/IR/ValueRange.h"             // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"              // from @llvm-project
+#include "mlir/include/mlir/Support/LogicalResult.h"     // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+#define DEBUG_TYPE "tosa-to-secret-arith"
+
+namespace mlir {
+namespace heir {
+namespace tosa {
+
+#define GEN_PASS_DEF_TOSATOSECRETARITH
+#include "lib/Conversion/TosaToSecretArith/TosaToSecretArith.h.inc"
+
+Value createConstantFloat(ImplicitLocOpBuilder &b, double floatValue,
+                          RankedTensorType type) {
+  SmallVector<APFloat> constantValues;
+  constantValues.reserve(type.getNumElements());
+
+  auto elementType = type.getElementType();
+
+  // Create APFloat based on the float type width (this can definitely be made
+  // more elegant somehow)
+  APFloat value(0.0);  // Default initialization
+  if (elementType.isF32()) {
+    value = APFloat(static_cast<float>(
+        floatValue));  // Convert double to float if necessary
+  } else if (elementType.isF64()) {
+    value = APFloat(floatValue);  // Use the double value directly
+  } else {
+    llvm_unreachable("Expected a valid float type for constant creation");
+  }
+
+  // Fill the constant vector with the same value
+  constantValues.assign(type.getNumElements(), value);
+
+  auto constantValuesAttr = DenseElementsAttr::get(type, constantValues);
```

Comment:
just a style thing - you can also instantiate a SplatElementsAttr with a single value rather than a vector of values

Reply:
Thanks for letting me know! Fixed it!